### ISSUE TITLE
`jsdoc/require-jsdoc` doesn't report on methods of an exported class

### ISSUE THREAD

**Issue by garyking** [gajus/eslint-plugin-jsdoc#519]
Settings: 

```json
{
  "jsdoc/require-jsdoc": [
    "warn",
    {
      "publicOnly": true,
      "require": {
        "ArrowFunctionExpression": true,
        "ClassDeclaration": false,
        "ClassExpression": false,
        "FunctionDeclaration": true,
        "FunctionExpression": true,
        "MethodDefinition": true
      }
    }
  ]
}
```

Code:

```js
export default class Test {
  constructor(a) {
    this.a = a;
  }
}
```

No error on `constructor`, unless `publicOnly` changes to `false`, even though all the methods in the class are exported.

On a side note, would be nice for the rule to be able to skip reports on `private` methods in TypeScript classes.

**Comment by brettz9** [comment]
Note that in 7f34ca4b91806a6426a8faf2f280e3405af328c1 I've added examples of how to skip reports on private methods. If necessary, you can adapt the approach further using esquery selectors: https://github.com/estools/esquery .

**Comment by arthurkhlghatyan** [comment]
@gajus It doesn't work for neither 23.0.1 nor 24.0.0

**Comment by brettz9** [comment]
@arthurkhlghatyan : What is your exact config and with what code are getting an error? Please produce a test case.

**Comment by arthurkhlghatyan** [comment]
OK guys for sure.

I have a simple class

`
class Main {
    sayHello() {
        console.log('Hello World');
    }
}
`

And I'm using recommended styleguide from jsdoc **plugin:jsdoc/recommended** also tried to enable jsdoc/require-jsdoc manually but doesn't work.

With setup everything is fine, cause I tested by creating a simple function for which eslint required doc comments, so I'm 100% sure configuration is fine.

I tried V23.0.1, V23.1.0 and V24.0.0 - the last one was picked by the package manager by default.

Please let me know if you need anything else

**Comment by arthurkhlghatyan** [comment]
@brettz9 Please let me know if you need anything else.

**Comment by brettz9** [comment]
Method definitions are not required by the default setting for the rule. You need to set:

```js
{
  rules: {
    "jsdoc/require-jsdoc": [
      "warn",
      {
        "publicOnly": true,
        "require": {
          "MethodDefinition": true
        }
      }
    ]
  }
}
```

See the README for the other options.

If you still have issues, please open a new issue and give the exact configuration you are using with the rule (as I did above).

**Comment by arthurkhlghatyan** [comment]
Thanks @brettz9, let me see.

**Comment by arthurkhlghatyan** [comment]
Yeah that works, thanks @brettz9.
